lib/index.js

```
@@ -15,329 +15,303 @@ const internals = {};
-
-    const migrationGroup = { models, migrationsDir, knex };
-
-    // Grabbed from Schwifty's migrator.js
-    // Absolutize and normalize
-
-    migrationGroup.migrationsDir = Path.isAbsolute(migrationGroup.migrationsDir) ?
-        Path.normalize(migrationGroup.migrationsDir) :
-        Path.join(process.cwd(), migrationGroup.migrationsDir);
-
-    // Grab join table info
-
-    let joinTableCollector = [];
+    // Ping the knex instance to make sure we're connected to the db
+    userOptions.knex.queryBuilder().select(userOptions.knex.raw('1'))
+        .then(() => {
 
-    Items.serial(migrationGroup.models, (model, next) => {
+            const { models, migrationsDir, knex } = userOptions;
+            const migrationGroup = { models, migrationsDir, knex };
 
-        internals.getJoinTableInfo(model, knex, (_, info) => {
+            // Grabbed from Schwifty's migrator.js
+            // Absolutize and normalize
 
-            if (info) {
-                joinTableCollector = joinTableCollector.concat(info);
-            }
-            next();
-        });
-    },
-    (_) => {
+            migrationGroup.migrationsDir = Path.isAbsolute(migrationGroup.migrationsDir) ?
+                Path.normalize(migrationGroup.migrationsDir) :
+                Path.join(process.cwd(), migrationGroup.migrationsDir);
 
-        const joinTables = joinTableCollector.reduce((collector, joinTable) => {
+            // Grab join table info
 
-            // Dedupe the joinTables -- multiple models can reference the same joinTable,
-            // so we might have more than one entry for the same join table
-            // across different models.
+            let joinTableCollector = [];
 
-            const { tableName, joiSchema } = joinTable;
+            Items.serial(migrationGroup.models, (model, next) => {
 
-            if (!collector[tableName]) {
+                internals.getJoinTableInfo(model, knex, (ignoreError, info) => {
```

I feel pretty sure we shouldn't be ignoring this error—it reaches out to the database, right?

---

Yes there are a few things that reach out to the db that I'm ignoring errors from: Main reason I wanted to move this forward and don't really know how to test those errors

---

You can move forward with missing test coverage, and we can work on how to best test that.  The danger of ignoring errors just for test coverage isn't worth it!

---

Alright 👍

---

I did some pretty wacky stuff in Tandy in order to trigger errors - you might find some inspiration there.